### PR TITLE
POR-2379 tail logs for a given app/service from the cli

### PR DESCRIPTION
**Pull request by ianedwards**
## 
## What does this PR do?

- Basic webhook client setup for CLI
- `porter app logs` command using new WH client to stream logs

### CODE REVIEW

File: api/client/api.go [hunk 1]
```diff
@@ -80,6 +82,65 @@ func NewClientWithConfig(ctx context.Context, input NewClientInput) (Client, err
...
+
+	// handle encoding of timestamps
+	encoder.RegisterEncoder(time.Time{}, func(t reflect.Value) string {
+		return t.Interface().(time.Time).Format(time.RFC3339)
+	})
+
+	vals := map[string][]string{}
+	err := encoder.Encode(data, vals)
+	if err != nil {
+		return conn, err
```
Comment (d-g-town) [inline api/client/api.go:110]: wrap these errors

File: api/client/porter_app.go [hunk 2]
```diff
@@ -262,6 +264,65 @@ func (c *Client) UpdateApp(
...
+
+	req := &porter_app.AppLogsRequest{
+		ServiceName:          inp.ServiceName,
+		DeploymentTargetName: inp.DeploymentTargetName,
+		StartRange:           inp.StartRange,
+	}
+
+	err := c.getRequest(
+		fmt.Sprintf(
+			"/projects/%d/clusters/%d/apps/%s/logs",
```
Comment (d-g-town) [inline api/client/porter_app.go:292]: if possible and not too much of a blocker, can we start using the deployment target scoped endpoint for all new api calls?

File: api/client/porter_app.go [hunk 3]
```diff
@@ -262,6 +264,65 @@ func (c *Client) UpdateApp(
...
+	inp AppLogsInput,
+) (*websocket.Conn, error) {
+	req := &porter_app.AppLogsRequest{
+		ServiceName:          inp.ServiceName,
+		DeploymentTargetName: inp.DeploymentTargetName,
+	}
+
+	conn, err := c.websocketDial(
+		fmt.Sprintf(
+			"/projects/%d/clusters/%d/apps/%s/logs/loki",
```
Comment (d-g-town) [inline api/client/porter_app.go:314]: ditto

File: cli/cmd/commands/app.go [hunk 4]
```diff
@@ -574,6 +587,36 @@ func appRollback(ctx context.Context, _ *types.GetAuthenticatedUserResponse, cli
...
+	serviceName := v2.ServiceName_AllServices
+	if serviceFlag != "" {
+		serviceName = serviceFlag
+	}
+
+	err = v2.AppLogs(ctx, v2.AppLogsInput{
+		CLIConfig:            cliConfig,
+		Client:               client,
+		AppName:              appName,
+		DeploymentTargetName: deploymentTargetName,
```
Comment (d-g-town) [inline cli/cmd/commands/app.go:610]: where does this deployment target name come from?

**Review by d-g-town — APPROVED** [review]

File: cli/cmd/commands/app.go (hunk 4 above)
Comment (ianedwards) [reply cli/cmd/commands/app.go:610]: it's the -x flag attached to all app subcommands

**Pull request merged**